File: src/python/sequencial.py
from random import randint

def produzir_dados():
    # return randint(0, 100)
    # []
    # preenchimento de 100 (range(100))
    # para cada execução na interação, randint(0, 110)
    # dados = []
    # for _ in range(100):
    #    dados.append(randint(0, 110))
    dados = [randint(0, 110) for _ in range(100)]
    return dados

def consumir_dados(dados):
    resultado = sum(dados)
    print(f'recebeu -> {resultado}')

def principal():
    print("iniciou")
    dados = produzir_dados()
    consumir_dados(dados)
    print("finalizou")

if __name__ == '__main__':
    principal()

File: src/cpp/sequencial.cpp
#include <iostream>
#include <random>
#include <vector>

using namespace std;


vector<int> produzir_dados() {
    vector<int> dados;

    random_device rd; // Gera uma semente para a geração dos números inteiros
    mt19937 gerador(rd()); // Gera os números aleatorios
    uniform_int_distribution<int> distribuicao(0, 110); // Transforma em uma sequência de 0 a 110

    for (int i = 0; i < 100; i++) {
        dados.push_back(distribuicao(gerador));
    }

    return dados;
}


void consumir_dados(vector<int> dados) {
    int resultado;
    for (int i = 0; i < 100; ++i) {
        resultado += dados[i];
    }
    cout << "Recebeu -> " << resultado << endl;
}
int main() {
    cout << "iniciou" << endl;
    vector<int> dados;
    dados = produzir_dados();
    consumir_dados(dados);
    cout << "finalizou" << endl;
    return 0;
}
